core/src/test/java/io/questdb/test/std/IntListTest.java

```diff
@@ -151,6 +154,43 @@ public void testSmoke() {
         Assert.assertEquals(0, list.size());
     }
 
+    @Test
```

Comment:
Need more tests.
* Should test with different sized arrays (inc non-3 multiples).
* Should test with groups where all values in the group are the same, e.g.
  ```
  [  0, 0, 0,
     1, 1, 1,
     2, 2, 2  ]
  ```
  This would test the `compareGroups` logic to all its iterations, rather than just one deep or max two deep as you're most likely to get with random numbers.

Reply:
I added a test that generates equal elements group, thanks for the suggestion and extended implementation beyond 3 elements in the sort group.

The sorting implementation class is not open as a public method, so there is no way to pass wrong group indices.